fbw-a32nx/src/systems/extras-host/modules/gsx_sync/GsxSync.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2021-2024 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+import { ConsumerSubject, DebounceTimer, EventBus, SimVarValueType } from '@microsoft/msfs-sdk';
+import { ExtrasSimVarEvents, GsxSimVarEvents, NXDataStore } from '@flybywiresim/fbw-sdk';
+
+export class GsxSync {
+  private readonly sub = this.bus.getSubscriber<ExtrasSimVarEvents & GsxSimVarEvents>();
+
+  private readonly DEFUEL_DIFF_TARGET = 125;
+
+  private readonly numberOfGPUs = 1;
+
+  private readonly ExtPowerAvailStates = new Map<number, ConsumerSubject<boolean>>();
+
+  private readonly stateJetway = ConsumerSubject.create(null, 1);
+
+  private readonly stateBoard = ConsumerSubject.create(null, 1);
+
+  private readonly stateDeparture = ConsumerSubject.create(null, 1);
+
+  private readonly stateGpu = ConsumerSubject.create(null, 1);
+
+  private isDefuel = false;
+
+  private readonly defuelTimer = new DebounceTimer();
+
+  private readonly rateTimer = new DebounceTimer();
+
+  private desiredFuel = 0;
+
+  private readonly fuelHose = ConsumerSubject.create(null, 0);
+
+  private readonly stateRefuel = ConsumerSubject.create(null, 1);
+
+  private readonly refuelStarted = ConsumerSubject.create(null, false);
+
+  constructor(private readonly bus: EventBus) {
+    for (let index = 1; index <= this.numberOfGPUs; index++) {
+      const element = ConsumerSubject.create(this.sub.on(`ext_power_available_${index}`), false);
+      this.ExtPowerAvailStates.set(index, element);
+    }
+  }
+
+  public connectedCallback(): void {
+    this.stateJetway.setConsumer(this.sub.on('gsx_jetway_state'));
+    this.stateBoard.setConsumer(this.sub.on('gsx_boarding_state'));
+    this.stateDeparture.setConsumer(this.sub.on('gsx_departure_state'));
+    this.stateGpu.setConsumer(this.sub.on('gsx_gpu_state'));
+    this.fuelHose.setConsumer(this.sub.on('gsx_fuelhose_connected'));
+    this.stateRefuel.setConsumer(this.sub.on('gsx_refuel_state'));
+    this.refuelStarted.setConsumer(this.sub.on('a32nx_refuel_started_by_user'));
+
+    this.fuelHose.sub(this.onFuelHoseConnected.bind(this));
+    this.stateRefuel.sub(this.onRefuelState.bind(this));
+    this.refuelStarted.sub(this.onRefuelStarted.bind(this));
+
+    this.stateBoard.sub(this.evaluateGsxPowerSource.bind(this));
+    this.stateJetway.sub(this.evaluateGsxPowerSource.bind(this));
+    this.stateGpu.sub(this.evaluateGsxPowerSource.bind(this));
+    this.stateDeparture.sub(this.evaluateGsxPowerSource.bind(this));
+  }
+
+  private onRefuelState(state: number): void {
+    if (state == 4) {
```

Comment:
Same, use enums for GSX states.

Reply:
enum'ed in fbw-common/src/systems/shared/src/GsxSync.ts